nca/NetworkConfig/NetworkConfigQueryRunner.py

```diff
@@ -120,106 +166,38 @@ def _execute_pair_configs_query(self, query_type, config1, config2, cmd_line_fla
         return query_to_exec.execute_and_compute_output_in_required_format(cmd_line_flag)
 
     def _run_query_for_each_config(self):
-        res, output, queries_not_executed = self._init_query_results_values()
+        query_result = QueryResult()
         for config in self.configs_array:
-            query_res, query_output, query_not_executed =\
-                self._execute_one_config_query(self.query_name, self._get_config(config))
-            res, output, queries_not_executed = \
-                self._update_query_results_after_one_iteration(res, query_res, output, query_output,
-                                                               queries_not_executed, query_not_executed)
-        return self._return_final_query_results(res, output, queries_not_executed)
+            query_result.update(self._execute_one_config_query(self.query_name, self._get_config(config)))
```

Comment:
Maybe the function `_execute_one_config_query` can also return a `QueryResult` object?
The input arg to `QueryResult.update()` can be a `QueryResult` object rather than a plain tuple.

Reply:
the tuple in the `QueryResult.update()` is actually returned from `NetworkConfigQuery.execute_and_compute_output_in_required_format`
(which is called and returned by `_execute_one_config_query `and _`execute_pair_configs_query `from this file).

it is not exactly the same as the `QueryResult` since:
1- the tuple includes a query result from one iteration while `QueryResult'`s goal is to include the "sum" of all iterations till now
2 - the tuple's "query_output" variable is `Union[str, dict]`, while the query_output of the `QueryResult` is  `list `that stores results from all different "iterations" 
3- while it is possible to add a new member to the `QueryResult `(output_from_one_iteration), it will be problematic to import it in the `NetworkConfigQuery ` to be returned (this will cause a circular import , since this file imports all from  `NetworkConfigQuery` )
and I think that it is not right to move the `QueryResult` class to the `NetworkConfigQuery.py `since it is used for calculations that are relevant only in this file (i.e. `NetworkConfigQueryRunner.py`))| (do you agree?)

So another approach to avoid using a tuple here, is to add another `dataClass ` in `NetworkConfigQuery.py` which will consist of the  returned values in `NetworkConfigQuery.execute_and_compute_output_in_required_format`
does this sounds good?

Reply:
ok, so let's just keep it as it is now.